libs/util/app-config/src/lib/app-config.ts

```diff
@@ -95,150 +84,99 @@ export function loadAppConfig() {
-        )
-      }
-
-      if (errors.length)
+      themeConfig =
+        parsedThemeSection === null
+          ? null
+          : ({
+              PRIMARY_COLOR: parsedThemeSection.primary_color,
+              SECONDARY_COLOR: parsedThemeSection.secondary_color,
+              MAIN_COLOR: parsedThemeSection.main_color,
+              BACKGROUND_COLOR: parsedThemeSection.background_color,
+              HEADER_BACKGROUND: parsedThemeSection.header_background,
+              TITLE_FONT: parsedThemeSection.title_font,
+              MAIN_FONT: parsedThemeSection.main_font,
+              FONTS_STYLESHEET_URL: parsedThemeSection.fonts_stylesheet_url,
+            } as ThemeConfig)
+
+      customTranslations = parseTranslationsConfigSection(
+        parsed,
+        'translations'
+      ) as CustomTranslationsAllLanguages
+
+      if (errors.length) {
         throw new Error(`One or more mandatory settings were missing from the configuration file.
 ${errors.join('\n')}`)
-      else if (warnings.length)
+      } else if (warnings.length) {
         console.warn(`One or more unexpected settings were encountered in the configuration file.
 ${warnings.join('\n')}`)
-
-      // will flatten nested objects using dotted properties
-      const flatten = (base, obj, isFirst) =>
-        Object.keys(obj).reduce((prev, curr) => {
-          const path = base ? `${base}.${curr}` : curr
-          const val = obj[curr]
-          if (isFirst) return { ...prev, [path]: flatten('', val, false) }
-          else if (typeof val === 'object')
-            return { ...prev, ...flatten(path, val, false) }
-          else return { ...prev, [path]: val }
-        }, {})
-
-      // flatten translations by language
-      const translations = translationsNested
-        ? flatten('', translationsNested, true)
-        : {}
-
-      globalConfig = {
-        GN4_API_URL: global.geonetwork4_api_url,
-        PROXY_PATH: global.proxy_path,
-      }
-      const layersConfig: LayerConfig[] = []
-      if (map_layer) {
-        map_layer.forEach((layerConfig) => {
-          layersConfig.push({
-            TYPE: layerConfig.type,
-            URL: layerConfig.url,
-            NAME: layerConfig.name,
-          })
-        })
-      }
-      mapConfig = {
-        MAX_ZOOM: map.max_zoom,
-        MAX_EXTENT: map.max_extent,
-        DO_NOT_USE_DEFAULT_BASEMAP: !!map.do_not_use_default_basemap,
-        MAP_LAYERS: layersConfig,
-      }
-      themeConfig = {
-        PRIMARY_COLOR: theme.primary_color,
-        SECONDARY_COLOR: theme.secondary_color,
-        MAIN_COLOR: theme.main_color,
-        BACKGROUND_COLOR: theme.background_color,
-        HEADER_BACKGROUND: theme.header_background,
-        TITLE_FONT: theme.title_font,
-        MAIN_FONT: theme.main_font,
-        FONTS_STYLESHEET_URL: theme.fonts_stylesheet_url,
       }
-      customTranslations = translations || {}
 
       appConfigLoaded = true
     })
```

Comment:
Do we actually still need the `_reset()` function below ?

Reply:
I think so, for tests in `app-config.spec`